packages/logging_cloudwatch/aws_logging_cloudwatch/lib/src/cloudwatch_logger_plugin.dart

```diff
@@ -289,7 +289,38 @@ class CloudWatchLoggerPlugin extends AWSLoggerPlugin
     if (!_enabled) {
       return false;
     }
+
     final loggingConstraint = _getLoggingConstraint();
+
+    String? userId;
+
+    Amplify.Hub.listen(HubChannel.Auth, (AuthHubEvent event) async {
+      if (event.type == AuthHubEventType.signedOut ||
+          event.type == AuthHubEventType.userDeleted ||
+          event.type == AuthHubEventType.sessionExpired) {
+        userId = null;
+      }
+      if (event.type == AuthHubEventType.signedIn) {
+        userId = event.payload?.userId;
+      }
+    });
+
+    if (loggingConstraint.userLogLevel.containsKey(userId)) {
+      final userLevel = loggingConstraint.userLogLevel[userId]!;
+
+      if (userLevel.categoryLogLevel.containsKey(logEntry.loggerName)) {
```

Comment:
loggerName and categoryLogLevel keys are not exact match. loggerName contains category name.
e.g: logger name for Storage category can be `AWS.Amplify.Storage` or `AWS.Amplify.Storage.AmplifyStorageS3Dart`.